pandas/core/indexes/interval.py

```diff
@@ -234,6 +234,7 @@ def _simple_new(cls, array, name, closed=None):
         result = IntervalMixin.__new__(cls)
         result._data = array
         result.name = name
+        result._no_setting_name = False
```

Comment:
does this need to be set to False in each _simple_new, or can it just use the class-level default?

Reply:
If it’s just on the class, won’t it be shared amongst all instances? I think I should change the type declaration to not have a default.

Reply:
> If it’s just on the class, won’t it be shared amongst all instances?

I think the relevant thing is: when you set `level._no_setting_name = True` in the one place where you do that, will that update the class attribute, and the answer should be no.

Reply:
Yeah, ``level._no_setting_name = True`` adds an instance attribute, so there shouldn't happen anything on the class attribute.